src/main/resources/templates/kotlin-lang/kotlinClassGraphqlRequest.ftl

```diff
@@ -0,0 +1,149 @@
+package ${package}
+
+</#if>
+import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLOperation
+import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLOperationRequest
+<#if toString || equalsAndHashCode>
+import java.util.Objects
+</#if>
+
+<#if javaDoc?has_content>
+/**
+<#list javaDoc as javaDocLine>
+ * ${javaDocLine}
+</#list>
+ */
+</#if>
+<#if generatedInfo.getGeneratedType()?has_content>
+@${generatedInfo.getGeneratedType()}(
+    value = ["com.kobylynskyi.graphql.codegen.GraphQLCodegen"],
+    date = "${generatedInfo.getDateTime()}"
+)
+</#if>
+<#list annotations as annotation>
+@${annotation}
+</#list>
+open class ${className}(private val alias: String?) : GraphQLOperationRequest {
+
+    companion object {
+        const val OPERATION_NAME: String = "${operationName}"
+        val OPERATION_TYPE: GraphQLOperation = GraphQLOperation.${operationType}
+<#if builder>
+
+        fun builder(): Builder = Builder()
+</#if>
+    }
+
+    private val input: MutableMap<String, Any?> = LinkedHashMap()
+
+    constructor(): this(null)
+
+<#if fields?has_content>
+<#list fields as field>
+<#if field.javaDoc?has_content>
+    /**
+<#list field.javaDoc as javaDocLine>
+     * ${javaDocLine}
+</#list>
+     */
+</#if>
+<#if field.deprecated>
+    @java.lang.Deprecated
+</#if>
+    fun set${field.name?cap_first}(${field.name}: ${field.type}) {
+        this.input["${field.originalName}"] = ${field.name}
+    }
+
+</#list>
+</#if>
+    override fun getOperationType(): GraphQLOperation = OPERATION_TYPE
+
+    override fun getOperationName(): String = OPERATION_NAME
+
+    override fun getAlias(): String? = alias ?: OPERATION_NAME
+
+    override fun getInput(): MutableMap<String, Any?> = input
+
+<#if equalsAndHashCode>
+    override fun equals(other: Any?): Boolean {
+        if (this === other) {
+            return true
+        }
+        if (other == null || javaClass != other.javaClass) {
+            return false
+        }
+        val that = other as ${className}
+        return Objects.equals(operationType, that.operationType) &&
+                   Objects.equals(operationName, that.operationName) &&
+                   Objects.equals(input, that.input)
+    }
+
+    override fun hashCode(): Int = Objects.hash(operationType, operationName, input)
+</#if>
+<#if toString>
+
+    override fun toString(): String = Objects.toString(input)
+</#if>
+
+<#if builder>
+    class Builder {
+
+        private var `$alias`: String? = null
+    <#if fields?has_content>
+        <#list fields as field>
+        <#if field.defaultValue?has_content>
+        private var ${field.name}: ${field.type} = ${field.defaultValue}
+        <#else>
+        <#if field.type?ends_with("?")>
+        private var ${field.name}: ${field.type} = null
+        <#else>
+        <#if MapperUtil.isKotlinPrimitive(field.type)>
+        <#assign default = MapperUtil.defaultValueKotlinPrimitive(field.type)/>
+        private var ${field.name}: ${field.type} = default
```

Comment:
If the field is mandatory and doesn't have a default value, why would we arbitrarily add a default just because the type happens to be a primitive? I don't think we should do that

Reply:
not find a good way to implement builder set..

Reply:
Isn't the last `else` part dealing with this already for any other type?

Reply:
lateinit cannot use in primitive types ，so set a default value for it. And  isKotlinPrimitive will be removed

Reply:
Then my apologies, I really didn't know that! So yeah the JVM details seem to leak into the language in some places. My bad.